--- a/web/modules/contrib/leaflet_more_maps/leaflet_more_maps.module
+++ b/web/modules/contrib/leaflet_more_maps/leaflet_more_maps.module
@@ -356,38 +356,110 @@
     ),
   );

-  $attr_thunderforest = 'Thunderforest <a href="http://thunderforest.com">OpenCycleMap</a>. ' . $attr_osm;
+  $attr_thunderforest = '<a href="https://www.thunderforest.com">Thunderforest</a>. ' . $attr_osm;
   $map_info['osm-cycle'] = array(
-    'label' => 'OSM Thunderforest Cycle (zoom 0..18)',
+    'label' => 'OSM Thunderforest Cycle (zoom 0..22)',
     'description' => t('OpenStreetMap for cyclists'),
     'settings' => $default_settings,
     'layers' => array(
       'layer' => array(
-        'urlTemplate' => $prot . "{s}.tile.opencyclemap.org/cycle/$zxy",
+        'urlTemplate' => $prot . "{s}.tile.thunderforest.com/cycle/$zxy",
         'options' => array('attribution' => $attr_thunderforest),
       ),
     ),
   );
   $map_info['osm-transport'] = array(
     // See http://thunderforest.com.
-    'label' => 'OSM Thunderforest Transport (zoom 0..18)',
+    'label' => 'OSM Thunderforest Transport (zoom 0..22)',
     'description' => t('OpenCycleMap with train & tram lines'),
     'settings' => $default_settings,
     'layers' => array(
       'layer' => array(
-        'urlTemplate' => $prot . "{s}.tile2.opencyclemap.org/transport/$zxy",
+        'urlTemplate' => $prot . "{s}.tile.thunderforest.com/transport/$zxy",
         'options' => array('attribution' => $attr_thunderforest),
       ),
     ),
   );
   $map_info['osm-landscape'] = array(
     // See http://thunderforest.com.
-    'label' => 'OSM Thunderforest Landscape (zoom 0..18)',
+    'label' => 'OSM Thunderforest Landscape (zoom 0..22)',
     'description' => t('OpenCycleMap with landscape'),
     'settings' => $default_settings,
     'layers' => array(
       'layer' => array(
-        'urlTemplate' => $prot . "{s}.tile3.opencyclemap.org/landscape/$zxy",
+        'urlTemplate' => $prot . "{s}.tile.thunderforest.com/landscape/$zxy",
+        'options' => array('attribution' => $attr_thunderforest),
+      ),
+    ),
+  );
+  $map_info['osm-outdoors'] = array(
+    // See http://thunderforest.com.
+    'label' => 'OSM Thunderforest Outdoors (zoom 0..22)',
+    'description' => t('OpenStreetMap for outdoor activities'),
+    'settings' => $default_settings,
+    'layers' => array(
+      'layer' => array(
+        'urlTemplate' => $prot . "{s}.tile.thunderforest.com/outdoors/$zxy",
+        'options' => array('attribution' => $attr_thunderforest),
+      ),
+    ),
+  );
+  $map_info['osm-transport-dark'] = array(
+    // See http://thunderforest.com.
+    'label' => 'OSM Thunderforest Transport Dark (zoom 0..22)',
+    'description' => t('A dark variant of OSM Thunderforest Transport'),
+    'settings' => $default_settings,
+    'layers' => array(
+      'layer' => array(
+        'urlTemplate' => $prot . "{s}.tile.thunderforest.com/transport-dark/$zxy",
+        'options' => array('attribution' => $attr_thunderforest),
+      ),
+    ),
+  );
+  $map_info['osm-spinal-map'] = array(
+    // See http://thunderforest.com.
+    'label' => 'OSM Thunderforest Spinal Map (zoom 0..22)',
+    'description' => t('A perfect map style for all your heavy-rock needs, based on OSM'),
+    'settings' => $default_settings,
+    'layers' => array(
+      'layer' => array(
+        'urlTemplate' => $prot . "{s}.tile.thunderforest.com/spinal-map/$zxy",
+        'options' => array('attribution' => $attr_thunderforest),
+      ),
+    ),
+  );
+  $map_info['osm-pioneer'] = array(
+    // See http://thunderforest.com.
+    'label' => 'OSM Thunderforest Pioneer (zoom 0..22)',
+    'description' => t('A turn-of-the-century railroad map style, evoking a sense of bygone mapping, based on OSM'),
+    'settings' => $default_settings,
+    'layers' => array(
+      'layer' => array(
+        'urlTemplate' => $prot . "{s}.tile.thunderforest.com/pioneer/$zxy",
+        'options' => array('attribution' => $attr_thunderforest),
+      ),
+    ),
+  );
+  $map_info['osm-mobile-atlas'] = array(
+    // See http://thunderforest.com.
+    'label' => 'OSM Thunderforest Mobile Atlas (zoom 0..22)',
+    'description' => t('A new map style by Thunderforest, based on OSM'),
+    'settings' => $default_settings,
+    'layers' => array(
+      'layer' => array(
+        'urlTemplate' => $prot . "{s}.tile.thunderforest.com/mobile-atlas/$zxy",
+        'options' => array('attribution' => $attr_thunderforest),
+      ),
+    ),
+  );
+  $map_info['osm-neighbourhood'] = array(
+    // See http://thunderforest.com.
+    'label' => 'OSM Thunderforest Neighbourhood (zoom 0..22)',
+    'description' => t('Another new map style by Thunderforest, based on OSM'),
+    'settings' => $default_settings,
+    'layers' => array(
+      'layer' => array(
+        'urlTemplate' => "https://tile.thunderforest.com/neighbourhood/$zxy?apikey=fe478abf78a944cc9f662fff1c8f520e",
         'options' => array('attribution' => $attr_thunderforest),
       ),
     ),

