diff --git a/src/drupal/web/modules/contrib/conditional_fields/src/DependencyHelper.php b/src/drupal/web/modules/contrib/conditional_fields/src/DependencyHelper.php
index 789f971..e99ea25 100644
--- a/src/drupal/web/modules/contrib/conditional_fields/src/DependencyHelper.php
+++ b/src/drupal/web/modules/contrib/conditional_fields/src/DependencyHelper.php
@@ -73,7 +73,7 @@ class DependencyHelper {
     if (!isset($this->dependencies[$this->entity_type][$this->bundle])) {
       $this->resolveBundleDependencies($this->getBundleDependentFields());
     }
-    return $this->dependencies[$this->entity_type][$this->bundle];
+    return $this->dependencies ? $this->dependencies[$this->entity_type][$this->bundle] : NULL;
   }
 
   /**
