--- a/Mapbox.php
+++ b/Mapbox.php
@@ -144,6 +144,11 @@
     private $geocodingMode;

     /**
+     * @var string
+     */
+    private $contextLocalityBuffer; // added via patch "mapbox-provider-locality.patch"
+
+    /**
      * @param HttpClient  $client        An HTTP adapter
      * @param string      $accessToken   Your Mapbox access token
      * @param string|null $country
@@ -312,6 +317,13 @@
             }
             $address = $address->withStreetName($builder->getValue('street_name'));
             $address = $address->withNeighborhood($builder->getValue('neighborhood'));
+
+            if (NULL !== $this->contextLocalityBuffer) {
+              // added via patch "mapbox-provider-locality.patch"
+              $address->subLocality = $this->contextLocalityBuffer;
+              $this->contextLocalityBuffer = NULL;
+            }
+
             $results[] = $address;

             if (count($results) >= $limit) {
@@ -341,6 +353,10 @@
                 break;

             case 'locality':
+               if (NULL === $this->contextLocalityBuffer) {
+                  // added via patch "mapbox-provider-locality.patch"
+                  $this->contextLocalityBuffer = $value['text'];
+                }
                 $builder->setLocality($value['text']);

                 break;
