--- a/src/drupal/core/modules/media/media.install
+++ b/src/drupal/core/modules/media/media.install
@@ -137,7 +137,7 @@
         ->getViewDisplay('media', $type->id());

       $source_field_definition = $type->getSource()->getSourceFieldDefinition($type);
-      if (!is_a($source_field_definition->getItemDefinition()->getClass(), ImageItem::class, TRUE)) {
+      if (empty($source_field_definition) || !is_a($source_field_definition->getItemDefinition()->getClass(), ImageItem::class, TRUE)) {
         continue;
       }
