--- a/web/modules/contrib/leaflet_more_maps/leaflet_more_maps.module
+++ b/web/modules/contrib/leaflet_more_maps/leaflet_more_maps.module
@@ -459,7 +459,7 @@
     'settings' => $default_settings,
     'layers' => [
       'layer' => [
-        'urlTemplate' => $prot . "{s}.tile.thunderforest.com/neighbourhood/$zxy",
+        'urlTemplate' => "https://tile.thunderforest.com/neighbourhood/$zxy?apikey=fe478abf78a944cc9f662fff1c8f520e",
         'options' => ['attribution' => $attr_thunderforest],
       ],
     ],
